app/controllers/posts_controller.rb

```diff
@@ -7,4 +7,53 @@ def show
     @post = Post.find_by(id: params[:id], author_id: params[:user_id])
     @user = User.find_by(id: params['user_id'])
   end
+
+  def new
+    @first_user = current_user
+    @post = Post.new
+  end
+
+  def create
+    @first_user = current_user
+    @post = Post.new(post_params.merge(author: @first_user))
+
+    if @post.save
+      redirect_to user_post_path(@first_user, @post)
+    else
+      render :new
+    end
+  end
+
+  def create_comment
+    @first_user = current_user
+    @post = Post.find(params[:id])
+    @comment = Comment.new(comment_params.merge(post: @post, author: @first_user))
+
+    if @comment.save
+      redirect_to user_post_path(@first_user, @post)
+    else
+      render :new_comment
+    end
+  end
+
+  def create_like
+    @post = Post.find(params[:id])
+    @like = Like.new(post: @post, author: current_user)
+
+    if @like.save
+      redirect_to user_post_path(current_user, @post)
+    else
+      render :new_comment
+    end
+  end
```

Comment:
- [ ] Instead of having create_like method here, please create a LikesController which should a create action 👍 
- [ ]  Please ensure to have a route in the config/routes.rb file for creating like that correspond with the create action